### PR TITLE
Added YoutubeVideo field

### PR DESCRIPTION
**Pull request by siyamandayubi**

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Views/VideoFieldSetting.Edit.cshtml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+<fieldset class="form-group">
+    <div class="row col-sm-6">
+        <label asp-for="Width">@T["Width"]</label>
+        <input type="number" asp-for="Width" class="form-control"/>
+        <span class="hint">@T["The width of the Viedobox"]</span>
+    </div>
+    <div class="row col-sm-6">
+        <label asp-for="Height">@T["Height"]</label>
+        <input type="number" asp-for="Height" class="form-control"/>
+        <span class="hint">@T["The height of the Viedobox"]</span>
```
Comment (Skrypt) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Views/VideoFieldSetting.Edit.cshtml]: Typo Viedobox = Videobox

**Comment by Skrypt** [comment]
Do we need the boolean for determining it's a Youtube video? Maybe if the intent is to define a different integration on frontend then we would need a list of different video streaming services instead to make this field more universal.

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Views/VideoFieldSetting.Edit.cshtml (hunk 1 above)
Comment (siyamandayubi) [reply src/OrchardCore.Modules/OrchardCore.ContentFields/Views/VideoFieldSetting.Edit.cshtml]: Thanks! 

I fixed it.

**Comment by siyamandayubi** [comment]
@Skrypt thanks for the tip. I change the checkbox to a combobox,  with two default streaming type: Self-hosted and Youtube (later we can simply extend them without breaking, since now it is a combobox). Getting rid of the configuration and checking the url address is also a good option in case it will work for the other streaming source that maybe we will add in the future.

**Comment by agriffard** [comment]
We should add oembed intégration like in O1. Paste url and detect oembed tags.

**Comment by siyamandayubi** [comment]
@sebastienros  
Please let me know that I got it correctly or not. I should change the driver of the field in such a way, that by simply pasting the youtube url into the editor, it should work? isn't it?

**Comment by siyamandayubi** [comment]
@sebastienros 
I took a look at the Drivers in OrchardCore. It would be good if next Tuesday, we talk about how to implement the DisplaySettings driver etc

**Comment by sebastienros** [comment]
Yes, the user should just paste the url of the main youtube video. Then on `UpdateAsync` you do a query to the oembed endpoint of youtube with this url, which will return a json document with the url you actually want.

**Comment by sebastienros** [comment]
https://oembed.com/

**Comment by agriffard** [comment]
May be `Embedded` instead of `Embeded`

**Pull request merged**